var/spack/repos/builtin/packages/gtkplus/package.py

```diff
@@ -60,4 +63,6 @@ def configure_args(self):
         args.append('GTKDOC_CHECK_PATH={0}'.format(true))
         args.append('GTKDOC_MKPDF={0}'.format(true))
         args.append('GTKDOC_REBASE={0}'.format(true))
+        if '~cups' in self.spec:
```

Comment:
https://github.com/spack/spack/pull/12655 has now been merged so I think this nearly all good as is. Generally for Autotools packages there is a corresponding `--enable-x` option for any `--disable-x` option - do you know if that applies to `--disable-cups`? If so, I think it would be worthwhile to add `--enable-cups` when `+cups` is specified.

Reply:
no there isn't:

gtk+-3.20.10$ ./configure --help | grep -e '--enable'
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --enable-dependency-tracking
  --enable-static[=PKGS]  build static libraries [default=no]
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-fast-install[=PKGS]
  --enable-debug=[no/minimum/yes]
  --enable-installed-tests
  --enable-xkb            support XKB extension [default=maybe]
  --enable-xinerama       support Xinerama extension [default=maybe]
  --enable-xrandr         support XRandR extension [default=maybe]
  --enable-xfixes         support XFixes extension [default=maybe]
  --enable-xcomposite     support X Composite extension [default=maybe]
  --enable-xdamage        support X Damage extension [default=maybe]
  --enable-x11-backend    enable the X11 gdk backend
  --enable-win32-backend  enable the Win32 gdk backend
  --enable-quartz-backend enable the quartz gdk backend
  --enable-broadway-backend
  --enable-wayland-backend
  --enable-mir-backend    enable the Mir gdk backend
  --enable-quartz-relocation
  --enable-explicit-deps=[yes/no/auto]
  --enable-test-print-backend
  --enable-introspection=[no/auto/yes]
  --enable-colord=[yes/no/auto]
  --enable-gtk-doc        use gtk-doc to build documentation [[default=no]]
  --enable-gtk-doc-html   build documentation in html format [[default=yes]]
  --enable-gtk-doc-pdf    build documentation in pdf format [[default=no]]
  --enable-man            generate man pages [default=auto]

gtk+-3.20.10$ ./configure --help | grep -e '--disable'

  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --disable-maintainer-mode
  --disable-dependency-tracking
  --disable-libtool-lock  avoid locking (might break parallel builds)
  --disable-largefile     omit support for large files
  --disable-glibtest      do not try to compile and run a test GLIB program
  --disable-modules       disable dynamic module loading
  --disable-cups          disable cups print backend
  --disable-papi          disable papi print backend
  --disable-cloudprint    disable cloudprint print backend
  --disable-schemas-compile
  --disable-doc-cross-references
  --disable-Bsymbolic     avoid linking with -Bsymbolic